File: hamcrest-php/Hamcrest/Matcher/String/IsEqualIgnoringCase.php
<?php
/*  Copyright (c) 2000-2008 hamcrest.org
 */
namespace Hamcrest;

/**
 * Tests if a string is equal to another string, regardless of the case.
 */
class IsEqualIgnoringCase extends BaseMatcher {
    protected $string;

    public function __construct($string) {
        if (!is_string($string)) {
            throw new IllegalArgumentException;
        }

        $this->string = $string;
    }

    public function matches($item) {
        return strtolower($this->string) === strtolower($item);
    }

    public function describeTo(Description $description) {
        $description->appendText('eqIgnoringCase(')
                    ->appendValue($this->string)
                    ->appendText(')');
    }
}

function equalToIgnoringCase($string) {
    return new IsEqualIgnoringCase($string);
}

File: hamcrest-python/hamcrest/library/text/isequal_ignoring_case.py
__author__ = "Jon Reid"
__copyright__ = "Copyright 2010 www.hamcrest.org"
__license__ = "BSD, see License.txt"
__version__ = "1.0"

from hamcrest.core.base_matcher import BaseMatcher


class IsEqualIgnoringCase(BaseMatcher):
    """Matches if strings are equal ignoring case."""

    def __init__(self, string):
        if not isinstance(string, str):
            raise TypeError('IsEqualIgnoringCase requires string')
        self.original_string = string
        self.lowered_string = string.lower()

    def _matches(self, item):
        if not isinstance(item, str):
            return False
        return self.lowered_string == item.lower()

    def describe_to(self, description):
        description.append_text('equal_to_ignoring_case(')  \
                    .append_value(self.original_string)     \
                    .append_text(')')


def equal_to_ignoring_case(string):
    """Are the strings equal, ignoring case?"""
    return IsEqualIgnoringCase(string)
